src/wrapper/simulation_objects.py

```diff
@@ -871,25 +907,20 @@ def update(self, data: dict):
             and not edge_id[:1] == ":"
         ):
             if hasattr(self, "_edge"):
-                try:
-                    assert self._edge.track.reservations[0][1] == self._edge
-                    assert self._edge.track == self.reserved_tracks[0]
-                    if len(self.reserved_tracks) > 1:
-                        assert (
-                            edge_id
-                            == self.reserved_tracks[1].reservations[0][1].identifier
-                        )
-                except Exception as exc:
-                    for track in self.reserved_tracks:
-                        print(track.edges[0].identifier)
+                if edge_id not in list(
+                    map(lambda obj: obj.identifier, self._edge.to_node.edges)
+                ):
                     raise ValueError(
                         (
                             "A Track was skipped: Old track: "
                             f"{self._edge.identifier}, new track: {edge_id}"
                         )
-                    ) from exc
-                self._edge.track.reservations.pop(0)
-                self.reserved_tracks.pop(0)
+                    )
+                if isinstance(self._edge.track, ReservationTrack):
```

Comment:
A cleaner solution probably would be one where the track has an attribute stating, if it is a ReservationTrack. But this may be fine too

Reply:
I changed that